app/src/main/java/org/kiwix/kiwixmobile/main/MainActivity.java

```diff
@@ -1963,4 +1975,54 @@ private void searchFiles() {
   public boolean checkNull(View view) {
     return view != null;
   }
+  public static void sendNotification(String message,Context context){
+
+        Intent actionIntent_stop = new Intent(context, NotificationService.class)
+                .setAction(ACTION_STOP);
+        PendingIntent actionPendingIntent_stop = PendingIntent.getService(context, 0,
+                actionIntent_stop, PendingIntent.FLAG_ONE_SHOT);
+        Intent actionIntent_pause_play = new Intent(context, NotificationService.class)
+                .setAction(ACTION_PAUSE);
+        PendingIntent actionPendingIntent_pause_play = PendingIntent.getService(context, 0,
+                actionIntent_pause_play, PendingIntent.FLAG_ONE_SHOT);
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(context)
+                .setSmallIcon(R.drawable.kiwix_icon_with_title)
+                .setContentTitle(context.getString(R.string.tts_stop))
+                .setVisibility(NotificationCompat.VISIBILITY_PUBLIC)
+                .addAction(R.drawable.ic_stop_black_24dp, context.getString(R.string.tts_stop),
+                        actionPendingIntent_stop)
+                .setPriority(Notification.PRIORITY_MAX);
+        if(message.equals(context.getString(R.string.tts_resume))) {
+            builder.addAction(R.drawable.ic_play_arrow_black_24dp, message,
+                    actionPendingIntent_pause_play);
+        }
+        else{
+            builder.addAction(R.drawable.ic_pause_black_24dp, message,
+                    actionPendingIntent_pause_play);
+        }
+        NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+        notificationManager.notify(TTS_NOTIFICATION_ID, builder.build());
+    }
+
+    public static class NotificationService extends IntentService {
+        public NotificationService() {
+            super(NotificationService.class.getSimpleName());
+        }
+        @Override
+        protected void onHandleIntent(Intent intent) {
+            String action = intent.getAction();
+            Log.d("TAG", "onHandleIntent: ");
+            if (action.equals(ACTION_STOP)) {
+                Log.d("TAG", "onHandleIntent: match");
+                MainActivity.tts.stop();
+                NotificationManagerCompat.from(this).cancel(001);
```

Comment:
Use the variable here too. Always re-check your code while making a fix. Reviewers might not be able to point out every single usage.

Reply:
But I have already fixed this. Check my previous commit :)